server/src/test/kotlin/com/handcontrol/server/grpc/GrpcTest.kt

```diff
@@ -0,0 +1,301 @@
+        assertThat(loginResponse.token != null)
+    }
+
+    @Test
+    fun loginUnAuth() {
+        val c = ManagedChannelBuilder.forTarget(target).usePlaintext().build();
+        channel = ClientInterceptors.intercept(c, authClientInterceptor)
+        val stub = HandleRequestGrpc.newBlockingStub(channel)
+
+        val exception: Exception = assertThrows(RuntimeException::class.java) {
+            stub.login(Request.LoginRequest.newBuilder().setLogin("noLogin").setPassword("noPass").build())
+        }
+        assertTrue(exception.message!!.contains("PERMISSION_DENIED"))
+    }
+
+    @Test
+    fun getOnline() {
+        val c = ManagedChannelBuilder.forTarget(target).usePlaintext().build();
+        channel = ClientInterceptors.intercept(c, authClientInterceptor)
+        val stub = HandleRequestGrpc.newBlockingStub(channel)
+
+        val loginResponse = stub.login(Request.LoginRequest.newBuilder().setLogin("login").setPassword("pass").build())
+
+        val signedStub = getSignedStub(loginResponse.token)
+
+        val response = signedStub.getOnline(Request.getOnlineRequest.newBuilder().build())
+        assertTrue(response!= null)
+    }
+
+    @Test
+    fun setOffline() {
+        val c = ManagedChannelBuilder.forTarget(target).usePlaintext().build();
+        channel = ClientInterceptors.intercept(c, authClientInterceptor)
+        val stub = HandleRequestGrpc.newBlockingStub(channel)
+
+        val loginResponse = stub.login(Request.LoginRequest.newBuilder().setLogin("login").setPassword("pass").build())
+
+        val signedStub = getSignedStub(loginResponse.token)
+
+        val response = signedStub.setOffline(Request.setOfflineRequest.newBuilder().setId("1").build())
+        assertTrue(response!= null)
+    }
+
+    @Test
+    fun getSettings() {
+        val c = ManagedChannelBuilder.forTarget(target).usePlaintext().build();
+        channel = ClientInterceptors.intercept(c, authClientInterceptor)
+        val stub = HandleRequestGrpc.newBlockingStub(channel)
+
+        val loginResponse = stub.login(Request.LoginRequest.newBuilder().setLogin("login").setPassword("pass").build())
+
+        val signedStub = getSignedStub(loginResponse.token)
+
+        val response = signedStub.getSettings(Request.getSettingsRequest.newBuilder().setId("1").build())
+        //println(response.settings)
+        assertTrue(response.settings!= null)
+    }
+
+    @Test
+    fun setSettings() {
+        val c = ManagedChannelBuilder.forTarget(target).usePlaintext().build();
+        channel = ClientInterceptors.intercept(c, authClientInterceptor)
+        val stub = HandleRequestGrpc.newBlockingStub(channel)
+
+        val loginResponse = stub.login(Request.LoginRequest.newBuilder().setLogin("login").setPassword("pass").build())
+
+        val signedStub = getSignedStub(loginResponse.token)
+
+        val settings = Settings.SetSettings.newBuilder()
+        settings.telemetryFrequency = 100
```

Comment:
use mockito everywhere instead of creating mocks

Reply:
what do you mean?